### ISSUE TITLE
Problem with triple curly braces between script and style tags

### ISSUE THREAD

**Issue by filipebraida** [sajansharmanz/prettier-plugin-edgejs#6]
I followed the migration guide that Virk posted on the blog for a new project I was working on and encountered the following issue:

Code:
```edge
  <script>
    window.addEventListener("load", function() {
      var datetimeField = document.getElementById("myDatetimeField");
      datetimeField.value = {{{ `"${timezone}"` }}};
    });
  </script>
  ```

Error:
```
[error] resources/file.edge: Error: {"message":"Unexpected token: punc «{», expected: punc «[»","filename":"file1.js","line":4,"col":29,"pos":150}
[error]     at wm (file:///home/filipebraida/Work/workspace/project/node_modules/prettier-edgejs/dist/main.js:8419:11)
```
  
I simplified the code a bit to make it easier to understand.

Thanks for the project!

**Comment by sajansharmanz** [comment]
Hey,

Thanks for the bug.

I'll have to look into this when I have some ample spare time. The issue is coming because between the script tag we are using minifyjs to format the javascript code. So the minifyjs library is throwing an error when trying to parse the edgejs related content.

Might be that I have to remove minifyjs in general to get this working, but will need further investigation.

Will let you know as soon as possible

**Comment by sajansharmanz** [comment]
@filipebraida could you please try the following for me - should fix your issue.

<script>
  window.addEventListener("load", function() {
    var datetimeField = document.getElementById("myDatetimeField");
    datetimeField.value = `{{timezone}}`
  });
</script>

**Comment by sajansharmanz** [comment]
@filipebraida could you please try the following for me - should fix your issue.


```
<script>
  window.addEventListener("load", function() {
    var datetimeField = document.getElementById("myDatetimeField");
    datetimeField.value = `{{timezone}}`
  });
</script>
```

You could use the triple curly brackets if you want html safe too
```
<script>
  window.addEventListener("load", function() {
    var datetimeField = document.getElementById("myDatetimeField");
    datetimeField.value = `{{{timezone}}}`
  });
</script>
```

The formatter won't break in this manner; and the EdgeJS will still parse the template correctly and pass in the variable as you need.

I'll continue my investigations until as well.

**Comment by sajansharmanz** [comment]
@thetutlage do you have any advice on how we should be passing variables from the controller to the EdgeJS template to be used in a script tag as such?

From my understanding we'd have to do one of the following;

1. Set a data attribute on an element and use JS to grab the value inside the script tag

2. JSON.stringify the data and then inside the script tag do something such as
```
JSON.parse(`{{{test1}}}`)
```

Is this correct?

**Comment by filipebraida** [comment]
Thank you for the response! I simplified the code to make the issue clearer.

I thought of a few ways to work around it, but I had to leave and haven’t had the chance to make any changes yet.

The problem is indeed more complicated, especially since it's important to format the JS code within Edge.

**Comment by filipebraida** [comment]
I fixed the code by moving the logic outside because any Edge constructor doesn’t work within the script tag.

This is my actual example with the fixed code:
```edge
  @if(textAnnotation)
    <script>
      window.addEventListener("load", function() {
        var datetimeField = document.getElementById("myDatetimeField");
        datetimeField.value = "{{ textAnnotation.entry.getStandardizedDate(timezone) }}";
      });
    </script>
  @else
    <script>
      window.addEventListener("load", function() {
        var datetimeField = document.getElementById("myDatetimeField");
        datetimeField.value = localDatetime();
      });
    </script>
  @end
  ```
  
  Original code:
  ```edge
  datetimeField.value = {{{ textAnnotation ? `"${textAnnotation.entry.getStandardizedDate(timezone)}"` : 'localDatetime()' }}};
  ```

**Comment by sajansharmanz** [comment]
Awesome - il keep this open for a bit longer in case Virk replies with any info that could improve the way we handle things.

Otherwise id like to update Edge doco + ReadMe file here explaining how it should he handled because im sure someone else will run into an issue eventually too

**Comment by filipebraida** [comment]
@sajansharmanz I also noticed that there are some code variations that trigger the same exception, even without any Edge code. In those cases, it was indeed incorrect code, but the error still appears. For example:

```edge
    <script>
      window.addEventListener("load", function() {
        var datetimeField = document.getElementById("myDatetimeField");
        datetimeField.value =
      });
    </script>
 ```
 
My question is whether this error should really appear even with incorrect code. If so, one option might be to make minifyjs optional, allowing the developer to enable it in the settings.

**Comment by thetutlage** [comment]
@filipebraida I think the behavior of throwing error is same with other files to when using Prettier. Prettier will show an error if the source file is invalid.

@sajansharmanz I think there are two things here. First, do we want to allow using Edge expression within the inline `script` and `style` tags. Technically they are allowed and should be allowed, even though I will recommend not using them heavily.

So yes, it will be nice if the printer can also check for Edge code within the `script` and the `style` tags. Or the printer can be limited to only the HTML portion and simply ignore inline styles and scripts.

**Comment by sajansharmanz** [comment]
Sounds good - I will disable our style and script formatters inside for now and investigate a way forward from there

Thanks!

**Comment by sajansharmanz** [comment]
@filipebraida I have published `prettier-edgejs@0.2.27` which disables the formatters I was using for style and script tags. The this will indent the tags and there content to the level they should sit, but it won't format the content inside anymore.

I will investigate a way forward from here - which might mean writing my own print logic for CSS and JS. I will keep this issue open as part of it and further updates.

**Comment by SUGEREK** [comment]
I have similar problem with prettier but in my case i don't get any errors from it. When im saving the file formatter simply removes one bracket from the end and seperate one from the beggining (im ataching photo). Problem only occurs when im trying to put 3 brackets inline of html object, outside of it its working normal.

![image](https://github.com/user-attachments/assets/1c880e85-60af-4ff4-b943-6d857b39fa41)

**Comment by filipebraida** [comment]
@SUGEREK I checked it here, and Prettier is indeed removing a closing bracket at the end. In fact, it’s also adding a space between the first two brackets and the third one.

**Comment by sajansharmanz** [comment]
@SUGEREK I will need to adjust the parser and formatter to handle triple bracket. In the meantime you can use double bracket or even an @if() tag as a workaround. 

I'll open another issue for this problem and move details there for it

**Comment by thetutlage** [comment]
Seems to be fixed https://github.com/sajansharmanz/prettier-plugin-edgejs/issues/8.

**Comment by sajansharmanz** [comment]
@filipebraida I have released prettier-edgejs@0.2.29 which re-enables the use of the <script> and <style> formatters. At the moment these tags will only accept mustache and safe mustache tags.

I tried to get @if, @else and @end to work - but its proving to be more complex. I will continue to investigate these when I get another chance.

**Comment by sajansharmanz** [comment]
@filipebraida I have released prettier-edgejs@0.2.29 which re-enables the use of the <script> and <style> formatters. At the moment these tags will only accept mustache and safe mustache tags.

I tried to get edge `@if`, `@else` and `@end` to work - but its proving to be more complex. I will continue to investigate these when I get another chance. Im going to close this issue and open another ticket to look at those specfically.
